### ISSUE TITLE
User-defined facet name on a type matches built-in facets

### ISSUE THREAD

**Issue by postatum** [aml-org/amf#479]
### Issue
Running code below should fail or report validation error because according to RAML 1.0 spec provided RAML input is invalid.

The spec says:
> User-defined facet names on a type MUST NOT match built-in facets on that type, nor facet names of any ancestor type in the inheritance chain of the type.

Spec link:
https://github.com/raml-org/raml-spec/blob/master/versions/raml-10/raml-10.md/#user-defined-facets

### AMF Version
4.0.0-SNAPSHOT

### Code
```sh
$ java -jar amf-4.0.0-SNAPSHOT.jar validate -in "RAML 1.0" -mime-in "application/yaml" -p "RAML" file://api.raml
{
  "@type": "http://www.w3.org/ns/shacl#ValidationReport",
  "http://www.w3.org/ns/shacl#conforms": true
}
```

### api.raml (facet of ancestor)
```raml
#%RAML 1.0
title: A

types:
  DateString:
    type: string
  Year:
    type: DateString
    facets:
      maxLength: number
  AnotherYear:
    type: Year
    maxLength: 5

/search:
  get:
    queryParameters:
      y: AnotherYear
```

### Another api.raml (facet of type)
```raml
#%RAML 1.0
title: A

types:
  DateString:
    type: string
    facets:
      maxLength: number
  Year:
    type: DateString
    maxLength: 4

/search:
  get:
    queryParameters:
      y: Year
```

**Comment by BlockLucas** [comment]
https://www.mulesoft.org/jira/browse/APIMF-1519

**Comment by gutee** [comment]
This is fixed and stated for 4.0.0 release
